src/coffee-script.coffee

```diff
@@ -62,7 +62,8 @@ exports.compile = compile = withPrettyErrors (code, options) ->
   for fragment in fragments
     # Update the sourcemap with data from each fragment
     if options.sourceMap
-      if fragment.locationData
+      # Do not include empty, whitespace, or semicolon-only fragments.
+      if fragment.locationData and not /^[;\s]*$/.test fragment.code
```

Comment:
That's fine. I didn't know whether we were creating good mappings for any of them, so that's why I asked.